packages/scripts/src/send-email-blast.js

```diff
@@ -0,0 +1,190 @@
+      let row = rowString.split(',');
+      return {
+        name: row[0].trim(),
+        email: row[1].trim(),
+        confirmation_id: row[2].trim(),
+      };
+    });
+}
+
+function printProgress(progress) {
+  process.stdout.clearLine();
+  process.stdout.cursorTo(0);
+  process.stdout.write(progress);
+}
+
+// remove duplicate email entries
+function findDuplicates(emails) {
+  let unique = [];
+  let result = [];
+  emails.forEach(email => {
+    if (!unique.includes(email.email)) {
+      unique.push(email.email);
+      result.push(email);
+    } else {
+      console.log('Duplicate found', JSON.stringify(email));
+    }
+  });
+  return result;
+}
+
+// get handlebar template
+// function getTemplate() {
+//   const templatePath = path.resolve(`${__dirname}/templates/email-blast-template.hbs`);
+
+//   const templateContent = fs.readFileSync(templatePath, 'utf-8');
+//   const template = handlebars.compile(templateContent, { strict: true });
+//   const body = template(mailable.context);
+//   return body;
+// }
+
+// send individual emails using AWS SES
+async function sendEmail(user) {
+  const { name, email, confirmation_id } = user;
+  // unsure of actual template body, using hardcoded html string for now
+  const mailOptions = {
+    from: process.env.MAIL_FROM,
+    to: [email],
+    subject: subjectOne,
+    body: templateOne,
+  };
+
+  if (!ses) return;
+
+  const params = {
+    Destination: {
+      ToAddresses: [...mailOptions.to],
+    },
+    Message: {
+      Body: {
+        Html: {
+          Charset: 'UTF-8',
+          Data: mailOptions.body,
+        },
+      },
+      Subject: {
+        Charset: 'UTF-8',
+        Data: mailOptions.subject,
+      },
+    },
+    //TO-DO: testing purposes, need to remove and replace with prod
+    Source: 'EHPRDoNotReply@dev.ehpr.freshworks.club',
+    // Source: process.env.MAIL_FROM || 'EHPRDoNotReply@dev.ehpr.freshworks.club',
+  };
+
+  try {
+    console.log('Trying to Send Email');
+    const mail = await ses.sendEmail(params).promise();
+    sent++;
+    return mail;
+  } catch (e) {
+    // set max retry count for failed emails
+    if (MAX_RETRY > retryCount) {
+      console.log(`Will try sending to ${user.email} ${MAX_RETRY - retryCount} more time(s)`);
+      retryCount++;
+      await sendEmail(user);
+    } else {
+      errors.push(user);
+      console.error(e.message);
+    }
+  }
+}
+
+// function to setup and send emails
+async function sendEmails(max) {
+  let emails = await getEmails();
+  console.log(`Sending ${emails.length} emails has begun...`);
+
+  // guard to limit emails and remove ones over max limit
+  // outputs to csv file
+  if (emails.length > max) {
+    generateEmailCSV(emails.slice(max), `out/unsent_emails${Date.now()}.csv`);
+    emails = emails.slice(0, max);
+  }
+
+  emails = findDuplicates(emails);
+  let start = new Date();
+
+  // Set up interval function to run multiple sends at a set rate.
+  const interval = setInterval(() => {
```

Comment:
I used `@supercharge/promise-pool` to control the number of concurrent execution when migrating users in HCAP